src/languageservice/utils.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import {
+    DidChangeLanguageFlavorParams,
+    LanguageFlavorChangedNotification,
+} from "../models/contracts/languageService";
+import StatusView from "../views/statusView";
+import { keywords } from "./nonTSqlKeywords";
+import SqlToolsServiceClient from "./serviceclient";
+
+export function getNonTSqlKeywords(): Set<string> {
+    return keywords;
+}
+
+export enum LanguageServiceOptions {
+    SwitchDisabled = 0,
+    SwitchEnabled = 1,
+    SwitchUnset = 2,
+}
+
+export function hasNonTSqlKeywords(
+    text: string,
+    nonTSqlKeywords: Set<string>,
+): boolean {

```

Comment:
For some reason, I envisioned this happening inside the Parser itself or within STS, rather than at the VS Code layer.  The Parser code is already parsing the entire document to identifying comments/literals/object names, and the STS code (see `HandleSyntaxParseRequest`) is enumerating errors.

I'm also concerned about the performance of using a slew of regexes on what could be a large script file.  Have you tried this with a long script (e.g. a deployment script)?